### DISCUSSION TITLE
postforward not work

### DISCUSSION THREAD

**Discussion by kinchan** [roehling/postsrsd#126] [Q&A]
Hi, first of all great thank you for project menbers.

I have problem, postforward not work, please help.

- postfix main.cf
```
recipient_canonical_maps = tcp:localhost:10002
recipient_canonical_classes= envelope_recipient,header_recipient
```
- postsrsd deamon
```
postsrsd -D -u mailnull -p /var/run/postsrsd/postsrsd.pid -d mail.example.com -s /usr/local/etc/postsrsd.secret -f 10001 -r 10002 -t 1800
```

- /etc/aliases
```
example: "|/usr/local/bin/postforward forwarder@example.com",\forwarder
```
- /var/log/maillog
```
Oct 18 05:43:56 mail postfix/local[82269]: A992C45B10B: to=<example@mail.example.com>, orig_to=<example@example.net>, relay=local, delay=0.94, delays=0.91/0/0/0.03, dsn=5.3.0, status=bounced (Command died with status 2: "/usr/local/bin/postforward forwarder@example.org". Command output: panic: runtime error: slice bounds out of range [:24] with length 21  goroutine 1 [running]: main.main()  github.com/zoni/postforward/postforward.go:170 +0xa5c )
```

postforward not work.

thank you for reading.

**Comment by roehling** [comment]
I don't know if any postforward users are reading the PostSRSd discussions here, but I am not involved in that project at all, so you may have more luck asking your questions [there](https://github.com/zoni/postforward/issues).

**Reply by kinchan** [reply]
thank you for reply.
ok, I will try postforward session.

**Comment by benchonaut** [comment]
( parser error in postforward )

in generic you should read this
https://blog.sys4.de/email-forward-mit-sieve-ohne-spf-dmarc-und-dkim-konflikte-de.html

the sieve approach is currently the only robust solution that is also generally accepted from gmail/O365 etc.

especially when using forwards from already-dkim-signed domains

( when using opendkim you can remove duplicates )
```
For example in OpenDKIM, 
there are parameter to remove old DKIM-Signature header when signing. 
That parameter is RemoveOldSignatures. 
```
[The documentation](http://www.opendkim.org/opendkim.conf.5.html) says that default value is no.
